### PR TITLE
Added function listViews()

### PR DESCRIPTION
**Pull request by FVallieres**
We want to use views with GLPI for reports purposes.

here is the link of the discussed topic :
[Topic discussing this idea](https://glpi.userecho.com/communities/1/topics/1076-using-db-views-in-glpi)
Q | A
-- | --
Bug fix? | no
New feature? | yes
BC breaks? | no
Deprecations? | no
Tests pass? | yes
Fixed tickets |

### CODE REVIEW

**Comment by trasher** [comment]
GLPI does not use views at all, I do not see why we should add this support; and proposal do not have any vote.

Also, this will probably cause issues when we'll try to add another SGBD engine.

So, :-1:  for me.

**Comment by orthagh** [comment]
I'm not against it, as i encountered the same issue in the past as a plugin developer.
Sometimes views can be helpful (still not recommended by the way as performance is usually bad).

And in fact, the syntax for listing views is pretty much the same in all sgbd we want (pg, sqlite), so ok for me

**Comment by trasher** [comment]
OK then. Anyways, tests are failing; and new added method must be tested.

File: inc/dbmysql.class.php [hunk 1]
```diff
@@ -970,8 +991,13 @@ public function tableExists($tablename, $usecache = true) {
...
       $retrieve_all = !$this->cache_disabled && empty($this->table_cache);
 
-      $result = $this->listTables($retrieve_all ? 'glpi_%' : $tablename);
       $found_tables = [];
+      $result = $this->listTables($retrieve_all ? 'glpi_%' : $tablename);
+      while ($data = $result->next()) {
+         $found_tables[] = $data['TABLE_NAME'];
+      }
+
+      $result = $this->listViews($retrieve_all ? 'glpi_%' : $tablename);
```
Comment (cedric-anne) [inline inc/dbmysql.class.php]: As views will be used only on specific case, I think this second request should only be done if necessary, i.e.
```php
// Search in view only if not found in base tables or if $retrieve_all flag is active
if ($retrieve_all || !in_array($tablename, $found_tables)) {
   ...
}
```

**Comment by FVallieres** [comment]
@orthagh We did not had time to address this issue. im closing the pr.

**Pull request closed without merging**